smslibrary/src/test/java/com/eis/smslibrary/SMSManagerTest.java

```diff
@@ -62,7 +71,7 @@ public void validMessage_isSent() {
         PowerMockito.mockStatic(SmsManager.class);
         when(SmsManager.getDefault()).thenReturn(managerMock);
 
-        managerInstance.sendMessage(VALID_MESSAGE);
+        MANAGER_INSTANCE.sendMessage(VALID_MESSAGE);
```

Comment:
This is not a [constant](https://web.archive.org/web/20120911192801if_/http://developers.sun.com/sunstudio/products/archive/whitepapers/java-style.pdf#G3.5938) and should not be all upper-case:
> The
following are considered to be constants:
>1. All static final primitive types (Remember that all interface fields are inherently static
final).
>2. **All static final object reference types that are never followed by "." (dot)**.
>3. All static final arrays that are never followed by "[" (dot).

Reply:
What do you mean?
`private static final SMSManager MANAGER_INSTANCE = SMSManager.getInstance();`

Reply:
In the line I mentioned, `MANAGER_INSTANCE` is followed by a dot. Therefore it's not an object reference type that is never followed by a dot, therefore it's not a constant.

Reply:
Where did you even find that definition?

Reply:
[Oh btw](https://docs.oracle.com/javase/specs/jls/se13/html/jls-6.html)

> Constant Names
> The names of constants in interface types should be, and final variables of class types may conventionally be, a sequence of one or more words, acronyms, or abbreviations, all uppercase, with components separated by underscore "_" characters. Constant names should be descriptive and not unnecessarily abbreviated. Conventionally they may be any appropriate part of speech.

>  final variables of class types may conventionally be a sequence of one or more words[...] all uppercase, with components separated by underscore "_" characters.

Reply:
Sun's coding style guide is from the year 2000, while the Oracle's language specification you found is a lot more recent, so I'm okay with going with that